src/statmodels.jl

```diff
@@ -460,6 +460,19 @@ struct PValue
 end
 PValue(p::PValue) = p
 
+float(p::PValue) = float(p.v)
+
+for op in [:(==), :≈, :<, :≤, :>, :≥, :(isless)] # isless and < to place nice with NaN
```

Comment:
I think you'll also want `isequal` for `NaN` handling. And I think the `isapprox` definition should be handled a bit differently, as we want to be able to forward keyword arguments as well (i.e. `atol`/`rtol`) when not using the infix shorthand.

Reply:
`isequal` seemed to work fine locally with `NaN`, but it's easy to add in and test. I've moved `isapprox` to handle the kwargs.